utiliti/src/de/gurkenlabs/utiliti/swing/StatusBar.java

```diff
@@ -26,42 +28,50 @@ private StatusBar() {
   }
 
   public static Container create() {
-    panel = new JPanel(new FlowLayout(FlowLayout.LEFT));
-
-    statusLabel = new JLabel("");
-    statusLabel.setFont(new Font(Style.FONTNAME_CONSOLE, Font.PLAIN, (int) (12 * Editor.preferences().getUiScale())));
-
-    zoomComboBox = new JComboBox<>(Zoom.getAll());
-    zoomComboBox.addItemListener(e -> {
-      if (settingZoom || e.getStateChange() != ItemEvent.SELECTED) {
-        return;
-      }
-
-      Zoom.set(((Zoom) zoomComboBox.getSelectedItem()).getValue());
-    });
-
-    panel.add(zoomComboBox);
-    panel.add(statusLabel);
+    try {
+      SwingUtilities.invokeAndWait(() -> {
```

Comment:
Wrapping with `invokeAndWait` is unnecessary here, As long as components aren’t connected to the component hierarchy they can be updated from any thread.

Reply:
Per the [swing documentation](https://docs.oracle.com/javase/tutorial/uiswing/concurrency/dispatch.html):

"Some Swing component methods are labelled "thread safe" in the API specification; these can be safely invoked from any thread. All other Swing component methods must be invoked from the event dispatch thread."

Component.add() is not labelled as thread safe, so it is not thread safe.

Reply:
Swing thread safety is only relevant while components are currently displayable i.e part of the component hierarchy of a`Window`. This is when swing has to guarantee that no changes to the components are made during painting and layout. Because this is not the case for disconnected components (e.g. there is no way for the RepaintManager to be aware the component even exists).

Also: What is done with the component created by this function. It is added to the visible component hierarchy by some other code, which then actually needs to happen on the EDT. It's the call-site of the factory method which has to ensure thread safety not the factory itself.

Moreover if `create()` method is called using `SwingUtilities.invokeLater()` (which it needs to if the created panels is added to a displayed component) then `SwingUtilities.invokeAndWait` will throw an `Error` (that's correct. Not an exception. An error)

Reply:
So, if I'm understanding correctly, we should be trying to make this thread safe here?

https://github.com/gurkenlabs/litiengine/blob/8783b07a9b2efb35fa21a3f74884e85c701f7bc8/utiliti/src/de/gurkenlabs/utiliti/Program.java#L39-L40

![image](https://user-images.githubusercontent.com/7155747/109403742-65013800-792d-11eb-9a2f-bb871cbf0b96.png)

Reply:
Yes that’s correct

Reply:
I didn't realize this a few days ago, but that method is already inside an invokeLater() block.